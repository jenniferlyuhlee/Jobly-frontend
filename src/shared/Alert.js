import React from "react";

/**
 * Alert component to display users of form submission result
 * - loops through messages and displays them
 * Props: type('danger', success'), messages []
 */
function Alert({type, messages = []}){
    return(
        <div className={`alert alert-${type}`} role="alert">
            <ul className="m-0">
            {messages.map(err => (
                <li className="small" key={err}>
                    {err}
                </li>
            ))}
            </ul>
        </div>
    )
}

export default Alert;