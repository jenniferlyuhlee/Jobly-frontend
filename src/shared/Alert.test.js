import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";

it ("renders without crashing", function () {
    render(<Alert />);
})

it("matches snapshot for error alerts", function() {
    let messages=["Password needs to be atleast 5 characters.", "Username taken."]

    const {asFragment}= render(
        <Alert type="danger" messages={messages}/>
    );
    expect(asFragment()).toMatchSnapshot();
})

it("matches snapshot for success alerts", function (){
    let messages=["Successful", "Saved"]

    const {asFragment} = render(
        <Alert type="success" messages={messages} />
    );
    expect(asFragment()).toMatchSnapshot();
});