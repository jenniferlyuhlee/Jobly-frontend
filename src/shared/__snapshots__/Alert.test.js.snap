// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for error alerts 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <ul
      class="m-0"
    >
      <li
        class="small"
      >
        Password needs to be atleast 5 characters.
      </li>
      <li
        class="small"
      >
        Username taken.
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success alerts 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <ul
      class="m-0"
    >
      <li
        class="small"
      >
        Successful
      </li>
      <li
        class="small"
      >
        Saved
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
